src/Clerk.ts

```diff
@@ -360,6 +355,19 @@ export default class Clerk {
           return next();
         }
 
+        // In development or staging environments only, based on the request's
+        // User Agent, detect non-browser requests (e.g. scripts). If there
+        // is no Authorization header, consider the user as signed out and
+        // prevent interstitial rendering
+        if (isDevelopmentOrStaging(this._restClient.apiKey) && !req.headers['user-agent']?.startsWith('Mozilla/')) {
```

Comment:
I'm worried that checking for "Mozilla" user agent might not be enough to guarantee this is a browser request. Why is this necessary?

Reply:
For browsers, almost 99.9% of the user-agent strings start with Mozilla. There are some exceptions, such as the old Opera version or the web browser of Apple TV v4. All in all, this is a cheap best-effort check to prevent rendering the interstitial if the HTTP Authorization header is empty. 

For example, if an app API is called from a CLI script, rendering the interstitial is confusing. With this fix, we will return a 401 instead.

Does that make sense?

Reply:
It does, thanks for explaining. There's certainly no guarantee. I guess the rationale is "it's fine since it's only for dev".